test/components/selectionBoxLayerTests.ts

```diff
@@ -1,193 +1,244 @@
 ///<reference path="../testReference.ts" />
 
-describe("SelectionBoxLayer", () => {
-  it("boxVisible()", () => {
-    let svg = TestMethods.generateSVG();
-    let sbl = new Plottable.Components.SelectionBoxLayer();
-    sbl.renderTo(svg);
+describe("Interactive Components", () => {
+  describe("SelectionBoxLayer", () => {
+
+    describe("Basic Usage", () => {
+      let svgWidth = 500;
+      let svgHeight = 500;
+
+      let svg: d3.Selection<void>;
+      let sbl: Plottable.Components.SelectionBoxLayer;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(svgWidth, svgHeight);
+        sbl = new Plottable.Components.SelectionBoxLayer();
+      });
+
+      it("can get and set the boxVisible() property", () => {
+        assert.strictEqual(sbl.boxVisible(), false, "The box is not visible by default");
+
+        sbl.renderTo(svg);
+        assert.strictEqual(sbl.boxVisible(), false, "The box is not visible by default after rendering to svg");
+
+        assert.strictEqual(sbl.boxVisible(true), sbl, "Setting the boxVisible attribute returns the selection box layer");
+        assert.strictEqual(sbl.boxVisible(), true, "Setting the boxVisible to true attribute works");
+        sbl.boxVisible(false);
+        assert.strictEqual(sbl.boxVisible(), false, "Setting the boxVisible attribute to false works");
+
+        svg.remove();
+      });
+
+      it("renders the box in accordance to boxVisible() property", () => {
+        sbl.renderTo(svg);
+
+        let selectionBox = svg.select(".selection-box");
+        assert.isTrue(selectionBox.empty(), "initilizes without box in DOM");
+
+        sbl.boxVisible(true);
+        selectionBox = svg.select(".selection-box");
+        assert.isFalse(selectionBox.empty(), "box is inserted in DOM when showing");
+
+        sbl.boxVisible(false);
+        selectionBox = svg.select(".selection-box");
+        assert.isTrue(selectionBox.empty(), "box is removed from DOM when not showing");
+
+        svg.remove();
+      });
+
+      it("does not error on destroy() when scales are not inputted", () => {
+        sbl.renderTo(svg);
+        assert.doesNotThrow(() => sbl.destroy(), Error, "can destroy even with no scales");
+
+        svg.remove();
+      });
```

Comment:
I really don't think this test should exist but should more be embedded into our unit tests or tested elsewhere?